scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala

```diff
@@ -298,20 +309,109 @@ object node {
+        required: Boolean,
+        initialValue: Option[FixedExpressionValue],
+        hintText: Option[String],
+        inputConfig: ParameterInputConfig,
+    ) {
+      def withName(name: String): FragmentParameter = copy(name = name)
+    }
+
+    @JsonCodec case class FixedExpressionValue(expression: String, label: String)
+
+    object ParameterInputMode extends Enumeration {
+      type ParameterInputMode = Value
+
+      implicit val typeEncoder: Encoder[ParameterInputMode.Value] =
+        Encoder.encodeEnumeration(ParameterInputMode)
+      implicit val typeDecoder: Decoder[ParameterInputMode.Value] =
+        Decoder.decodeEnumeration(ParameterInputMode)
+
+      val InputModeAny: Value                = Value("InputModeAny")
+      val InputModeAnyWithSuggestions: Value = Value("InputModeAnyWithSuggestions")
+      val InputModeFixedList: Value          = Value("InputModeFixedList")
+    }
+
+    object FragmentParameter {
+
+      implicit def encoder: Encoder[FragmentParameter] = deriveConfiguredEncoder[FragmentParameter]
+
+      private val fieldNameRequired     = "required"
+      private val fieldNameInitialValue = "initialValue"
+      private val fieldNameHintText     = "hintText"
+      private val fieldNameInputConfig  = "inputConfig"
+
+      private val defaultInputConfig: Json = Json.fromJsonObject(
+        JsonObject(
+          "inputMode"       -> Json.fromString(InputModeAny.toString),
+          "fixedValuesList" -> Json.Null
+        )
+      )
+
+      private val defaultNewFieldValues = Map(
+        fieldNameRequired     -> Json.fromBoolean(false),
+        fieldNameInitialValue -> Json.Null,
+        fieldNameHintText     -> Json.Null,
+        fieldNameInputConfig  -> defaultInputConfig
+      )
+
+      private def setDefaultIfAbsent(obj: JsonObject, fieldName: String): JsonObject =
+        if (!obj.contains(fieldName)) {
+          obj.add(fieldName, defaultNewFieldValues(fieldName))
+        } else {
+          obj
+        }
+
+      private def setAbsentNewFieldsToDefaults(aCursor: ACursor): ACursor = {
+        aCursor.withFocus(_.mapObject { jsonObject =>
+          List(fieldNameRequired, fieldNameInitialValue, fieldNameHintText, fieldNameInputConfig)
+            .foldLeft(jsonObject)((acc, fieldName) => setDefaultIfAbsent(acc, fieldName))
+        })
+      }
+
+      // Needed for compatibility
+      implicit def decoder: Decoder[FragmentParameter] =
+        deriveConfiguredDecoder[FragmentParameter].prepare(setAbsentNewFieldsToDefaults)
+
+      def apply(name: String, typ: FragmentClazzRef): FragmentParameter = {
+        FragmentParameter(
+          name,
+          typ,
+          required = false,
+          initialValue = None,
+          hintText = None,
+          inputConfig = ParameterInputConfig(InputModeAny, None)
+        )
+      }
+
+    }
+
+    @JsonCodec
+    case class ParameterInputConfig(
+        inputMode: ParameterInputMode.Value,
+        fixedValuesList: Option[
```

Comment:
Input mode is `FixedList`, but field is `fixedValuesList` - I think this should be unified to avoid confusion.